src/program/latex/mod.rs

```diff
@@ -969,149 +795,83 @@ impl Latex for Program {
-                                        wire.set_param(
-                                            expression,
-                                            diagram.column,
-                                            diagram.settings.texify_numerical_constants,
-                                        );
-                                    }
-                                }
-
-                                // update the gate name based on the modifiers
-                                let gate_name = diagram.set_modifiers(&gate, &mut wire);
-
-                                if let Some(_) = diagram.circuit.get(&qubit) {
-                                    // has ctrl gate, must identify ctrls and targs after filling circuit
-                                    if gate_name.starts_with('C') {
-                                        has_ctrl_targ = true;
-                                    }
-
-                                    // add the gate to the wire at column 0
-                                    wire.gates.insert(diagram.column, gate_name);
-                                }
-
-                                // push wire to diagram circuit
-                                diagram.push_wire(wire)?;
-                            }
-                            _ => (),
-                        }
-                    }
-
-                    diagram.column += 1;
-                }
-                // do nothing for all other instructions
-                _ => (),
+            if let Qubit::Fixed(name) = qubit {
+                let wire = Wire {
+                    ..Default::default()
+                };
+                diagram.circuit.insert(*name, Box::new(wire));
             }
         }
 
         // are implicit qubits required in settings and are there at least two or more qubits in the diagram?
         if diagram.settings.impute_missing_qubits {
             // add implicit qubits to circuit
-            diagram
-                .settings
-                .impute_missing_qubits(diagram.column, &mut diagram.circuit);
+            RenderSettings::impute_missing_qubits(instructions.len() as u32, &mut diagram.circuit);
         }
 
-        // only call method for programs with control and target gates
-        if has_ctrl_targ {
-            // identify control and target qubits
-            diagram.set_ctrl_targ()?;
+        for instruction in instructions {
+            // set QW for any unused qubits in this instruction
+            diagram.set_empty(&qubits, &instruction);
+
+            // parse gate instructions into a new circuit
+            if let Instruction::Gate(gate) = instruction {
+                diagram.parse_gate(&gate)?;
+                diagram.locate_targ()?;
+                diagram.column += 1;
+            } else if let Instruction::GateDefinition(_) = instruction {
+                // GateDefinition is supported and parsed in Gate
```

Comment:
Does this mean that all `GateDefinition`s are turned into `Gate` elsewhere? Should indicate the function name/path for easier reference.

Reply:
I'm not sure where this happens but somewhere when Quil is parsed in quil-rs `DEFGATE` is turned into `GATE`.

Reply:
@Shadow53 I made this commit which maybe clarifies somethings a bit https://github.com/rigetti/quil-rs/commit/30169aa43747428105f9cd6e9ffb0220a514b0cd